src/vmm/src/vmm_config/memory_hotplug.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+
+use crate::devices::virtio::mem::{VIRTIO_MEM_DEFAULT_BLOCK_SIZE, VIRTIO_MEM_DEFAULT_SLOT_SIZE};
+use crate::utils::bytes_to_mib;
+
+/// Errors associated with memory hotplug configuration.
+#[derive(Debug, thiserror::Error, displaydoc::Display)]
+pub enum MemoryHotplugConfigError {
+    /// Block size must not be lower than {0} MiB
+    InvalidBlockSizeTooSmall(usize),
+    /// Block size must be a power of 2
+    InvalidBlockSizePowerOfTwo,
+    /// Slot size must not be lower than {0} MiB
+    InvalidSlotSizeTooSmall(usize),
+    /// Slot size must be a multiple of block size ({0} MiB)
+    InvalidSlotSizeMultiple(usize),
+    /// Total size must not be lower than slot size ({0} MiB)
+    InvalidTotalSizeTooSmall(usize),
+    /// Total size must be a multiple of slot size ({0} MiB)
+    InvalidTotalSizeMultiple(usize),
+}
```

Comment:
I think we can drop `Invalid` from all of these (or at least from the ones that go `TooSmall` at the end, because that already implies its invalid imo)